lm_eval/api/model.py

```diff
@@ -304,16 +303,11 @@ def _encode_pair(self, context, continuation):
             continuation = context[-n_spaces:] + continuation
             context = context[:-n_spaces]
 
-        if self.AUTO_MODEL_CLASS == transformers.AutoModelForCausalLM:
-            whole_enc = self.tok_encode(context + continuation)
-            context_enc = self.tok_encode(context)
+        whole_enc = self.tok_encode(context + continuation)
+        context_enc = self.tok_encode(context)
 
-            context_enc_len = len(context_enc)
-            continuation_enc = whole_enc[context_enc_len:]
-
-        elif self.AUTO_MODEL_CLASS == transformers.AutoModelForSeq2SeqLM:
-            context_enc = self.tok_encode(context)
-            continuation_enc = self.tok_encode(continuation)
+        context_enc_len = len(context_enc)
+        continuation_enc = whole_enc[context_enc_len:]
```

Comment:
For T5, the context would need to be encoded separately since it uses EOS.

I can make some adjustments from this PR.